### PR TITLE
Remove libexecinfo on Alpine

### PR DESCRIPTION
**Pull request by morrisonlevi**
### Description

Fix GH-1824. The quick summary is Alpine 3.17 removed this library, and on older versions it's apparently not great quality anyway.

The biggest negative impact of this change is that when `DD_TRACE_HEALTH_METRICS_ENABLED` or `DD_LOG_BACKTRACE` or their INI equivalents are used, and a `SIGSEGV` occurs, that a backtrace will no longer be displayed. In other words, although this is a breaking change for Alpine platforms, the impact is quite small.

On the other hand, it has one less runtime dependency; Alpine users will no longer need to `apk add libexecinfo` before installing the tracer.

### Readiness checklist
- [x] Changelog has been added to the release document.
- [x] Tests added for this feature/bug.

### Reviewer checklist
- [x] Appropriate labels assigned.
- [x] Milestone is set.

### CODE REVIEW

**Review by realFlowControl — APPROVED** [review]
Looks good to me, but I think @bwoebi / @krakjoe should have a look before merging

**Comment by r1viollet** [comment]
can you see if the musl-dev package does not have the backtrace api ?

File: dockerfiles/verify_packages/installer/test_install_on_plesk.sh [hunk 1]
```diff
@@ -5,11 +5,14 @@ set -e
...
-/opt/plesk/php/7.4/bin/php ./build/packages/datadog-setup.php --php-bin=all
+/opt/plesk/php/8.0/bin/php ./build/packages/datadog-setup.php --php-bin=all
 
-assert_ddtrace_version "${new_version}" /opt/plesk/php/7.4/bin/php
-assert_ddtrace_version "${new_version}" /opt/plesk/php/7.4/sbin/php-fpm
+# Around 2022-12-20, plesk/plesk:18.0 was updated and the PHP 7.4 install
+# was no longer present. I don't know if this is a temporary regression, or if
+# it's purposefully pulled since PHP 7.4 went EOL.
+#assert_ddtrace_version "${new_version}" /opt/plesk/php/7.4/bin/php
+#assert_ddtrace_version "${new_version}" /opt/plesk/php/7.4/sbin/php-fpm
```
Comment (bwoebi) [inline dockerfiles/verify_packages/installer/test_install_on_plesk.sh:16]: I'd definitely bet on it being on purpose :-) Guess we can just remove it.

**Review by bwoebi — APPROVED** [review]
LGTM :-)

**Pull request merged**